pydantic/_internal/_decorators.py

```diff
@@ -521,6 +529,28 @@ def update_from_config(self, config_wrapper: ConfigWrapper) -> None:
             computed_field_dec.info._update_from_config(config_wrapper, name)
 
 
+def is_method_overridden(model_dc: type[Any]) -> tuple[bool, set[Any] | None]:
+    """Check if there are overridden methods in the decorator infos
+    Args:
+        model_dc: the class that called the decorator
+        res: the decorator infos
+    Returns:
+        a tuple of (bool, set[Any]|None)
+        bool: whether there are overridden methods
+        set[Any]|None: the overridden methods
+    """
+    registered_decorators = model_dc.__pydantic_decorators__
+    registered_validators = list(registered_decorators.validators.values())
+    registered_validators.extend(list(registered_decorators.field_validators.values()))
+    registered_validators.extend(list(registered_decorators.root_validators.values()))
+    registered_validators.extend(list(registered_decorators.model_validators.values()))
+    functions = [validator.func for validator in registered_validators if hasattr(validator.func, '__self__')]
+    if len(functions) != len(set(functions)):
```

Comment:
We can drop this comparison and just return the list of duplicates below, so we don't turn the list into a set twice.